From 82f0565c3c4922727fe3ff33e606d00e32b2da03 Mon Sep 17 00:00:00 2001
From: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
Date: Wed, 22 Mar 2023 12:29:43 +0530
Subject: [PATCH] wifi: ath12k: Modify the threshold param for Rx

BATCH_THRESHOLD and TIMER_THRESHOLD are modified to 64 and 200
for Rx.

Avoid HAL_SRNG_FLAGS_LOW_THRESH_INTR_EN for the RXDMA BUFFER ring type.

Signed-off-by: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp.c  | 11 ++++++++---
 drivers/net/wireless/ath/ath12k/hal.h |  4 ++--
 2 files changed, 10 insertions(+), 5 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/dp.c
+++ b/drivers/net/wireless/ath/ath12k/dp.c
@@ -358,13 +358,18 @@ int ath12k_dp_srng_setup(struct ath12k_b
 		params.intr_timer_thres_us = HAL_SRNG_INT_TIMER_THRESHOLD_RX;
 		break;
 	case HAL_RXDMA_BUF:
+		params.intr_batch_cntr_thres_entries = 0;
+		params.intr_timer_thres_us = HAL_SRNG_INT_TIMER_THRESHOLD_RX;
+		break;
 	case HAL_RXDMA_MONITOR_BUF:
 	case HAL_RXDMA_MONITOR_STATUS:
-		params.low_threshold = num_entries >> 3;
-		if (type == HAL_RXDMA_MONITOR_BUF) {
+		if (type == HAL_RXDMA_MONITOR_BUF)
 			params.low_threshold = DP_RX_MONITOR_BUF_LOW_TH;
-		}
+		else
+			params.low_threshold = num_entries >> 3;
+
 		params.flags |= HAL_SRNG_FLAGS_LOW_THRESH_INTR_EN;
+
 		params.intr_batch_cntr_thres_entries = 0;
 		params.intr_timer_thres_us = HAL_SRNG_INT_TIMER_THRESHOLD_RX;
 		break;
--- a/drivers/net/wireless/ath/ath12k/hal.h
+++ b/drivers/net/wireless/ath/ath12k/hal.h
@@ -802,13 +802,13 @@ struct hal_srng {
 
 /* Interrupt mitigation - Batch threshold in terms of numer of frames */
 #define HAL_SRNG_INT_BATCH_THRESHOLD_TX 64
-#define HAL_SRNG_INT_BATCH_THRESHOLD_RX 128
+#define HAL_SRNG_INT_BATCH_THRESHOLD_RX 64
 #define HAL_SRNG_INT_BATCH_THRESHOLD_PPE2TCL 0
 #define HAL_SRNG_INT_BATCH_THRESHOLD_OTHER 1
 
 /* Interrupt mitigation - timer threshold in us */
 #define HAL_SRNG_INT_TIMER_THRESHOLD_TX 200
-#define HAL_SRNG_INT_TIMER_THRESHOLD_RX 500
+#define HAL_SRNG_INT_TIMER_THRESHOLD_RX 200
 #define HAL_SRNG_INT_TIMER_THRESHOLD_PPE2TCL 30
 #define HAL_SRNG_INT_TIMER_THRESHOLD_OTHER 256
 
